File: Python/Home_tasks/Grocery_app/Grocery_app_test.py
import unittest
from Grocery_app_functions import  *
class TestGroceryManager(unittest.TestCase):
	def test_that_grocery_list_is_empty(self):
		items = []
		actual = show_all_available_items(items)
		expected ="Your grocery list is empty"
		self.assertEqual(actual,expected)

	def test_that_grocery_list_is_empty_two(self):
		items = []
		actual = size_of_items(items)
		self.assertEqual(actual,0)


	def test_you_can_add_an_item_to_the_grocery_list(self):
		groceries = []
		grocery = "Missile launcher"
		actual = add_items(grocery, groceries)
		expected = "Missile launcher successfully added to your grocery list"
		self.assertEqual(actual,expected)

	def test_that_grocery_you_can_add_more_than_one_item_grocery_list(self):
		groceries = []
		grocery = "Magnum_97k"
		groceries.append(grocery)
		groceries.append("Onions")
		actual = size_of_items(groceries)
		self.assertEqual(actual,2)

	def test_if_items_added_to_the_grocery_list_actually_exist_in_the_list(self):
		groceries = []
		grocery = "AK-47"
		add_items(grocery, groceries)
		actual = size_of_items(groceries)
		self.assertEqual(actual,1)



	def test_you_can_remove_an_item_from_the_grocery_list(self):
		groceries = ["MK-11","B1 Grenade"]
		actual = remove_items("MK-11", groceries)
		expected = "MK-11 successfully removed"
		self.assertEqual(actual,expected)

	def test_you_can_only_remove_an_existing_item_from_the_grocery_list(self):
		groceries = ["K-9", "AK-47"]
		grocery = "K-22"
		actual = remove_items(grocery, groceries)
		expected = "K-22 not in the grocery list"
		self.assertEqual(actual,expected)

File: java/Home_tasks/Grocery_app/GroceryAppTest.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;
import static org.junit.jupiter.api.Assertions.assertEquals;
import java.util.ArrayList;


	public class GroceryAppTest{

		GroceryApp groceryApp;

		@BeforeEach
		void setup (){

			groceryApp = new GroceryApp();
		}

		@Test
		public  void testThatGroceryListIsEmpty(){
			int actual = groceryApp.totalAmountOfItems();
			assertEquals(actual, 0);
	
			}
		@Test
		public void testThatAnItemCanBeAddedToTheGroceryList(){

	
			String response = groceryApp.addItems("Tomatoes");

			assertEquals(response, "Tomatoes have been successfully added");


			}


		@Test
		public void testThatAnItemCanBeStoredInTheGroceryList(){
	
			groceryApp.addItems("Tomatoes");
			int response = groceryApp.totalAmountOfItems();
			assertEquals(response, 1);


			}
		
		@Test
		public void testThatMoreThanOneItemCanBeStoredInGroceryList(){
	
			groceryApp.addItems("K9");
			groceryApp.addItems("B1 Grenade");
			int response = groceryApp.totalAmountOfItems();
			assertEquals(response, 2);


			}
	

	/*	@Test
		public void testThatOneItemCannotBeAddedToTheGroceryListTwice(){
	
			groceryApp.addItems("tomatoes");
			String response = groceryApp.addItems("tomatoes");
			assertEquals(response, "An Item cannot be entered twice");


			}

		
		@Test
		public void testThatOneItemWasAddedTwiceAndTheTotalNumberOfItemsIsOne(){
	
			groceryApp.addItems("tomatoes");
			groceryApp.addItems("tomatoes");
			int response = groceryApp.totalGrocerecies();
			assertEquals(response, 1);


			}
		
*/
		@Test
		public void testThatItemCanBeRemoveFromTheGroceryList(){
		
			groceryApp.addItems("H1 Bomb");
			String response = groceryApp.removeItems("H1 Bomb");
			assertEquals(response, "H1 Bomb have been successfully removed from the grocery list");
			}

	


		@Test
		public void testThatOnlyItemInTheGroceryListCanBeCanBeRemove(){
	
			groceryApp.addItems("Tomatoes");
			String response = groceryApp.removeItems("Fish");
			assertEquals(response, "An Item that is not on the grocery list cannot be removed");


			}





		@Test
		public void testThatYouCanViewItemsInTheGroceryList(){
	
			groceryApp.addItems("Tomatoes");
		
			ArrayList <String> result = groceryApp.showItems();
			String actual = result.get(0);
			String expected = "Tomatoes";
			assertEquals(actual, expected);


			}

	
	



}
